unit-test/broker-protection.js

```diff
@@ -556,25 +556,32 @@ describe('Actions', () => {
             )
         })
     })
+})
 
-    describe('fillForm', () => {
-        describe('generateRandomPhoneNumber', () => {
-            it('generates a string of integers of an appropriate size', () => {
-                const phoneNumber = generatePhoneNumber()
+describe('generators', () => {
+    describe('generateRandomPhoneNumber', () => {
+        it('generates a string of integers of an appropriate size', () => {
+            const phoneNumber = generatePhoneNumber()
 
-                expect(typeof phoneNumber).toEqual('string')
-                expect(phoneNumber.length).toBe(10)
-                expect(phoneNumber).toMatch(/^\d{10}$/)
-            })
+            expect(typeof phoneNumber).toEqual('string')
+            expect(phoneNumber.length).toBe(10)
+            expect(phoneNumber).toMatch(/^\d{10}$/)
         })
-        describe('generateZipCode', () => {
-            it('generates a string of integers of an appropriate size', () => {
-                const zipCode = generateZipCode()
+    })
+    describe('generateZipCode', () => {
+        it('generates a string of integers of an appropriate size', () => {
+            const zipCode = generateZipCode()
 
-                expect(typeof zipCode).toEqual('string')
-                expect(zipCode.length).toBe(5)
-                expect(zipCode).toMatch(/^\d{5}$/)
-            })
+            expect(typeof zipCode).toEqual('string')
+            expect(zipCode.length).toBe(5)
+            expect(zipCode).toMatch(/^\d{5}$/)
+        })
+    })
+    describe('generateStreetAddress', () => {
+        it('generates a string of integers of an appropriate size', () => {
+            const streetAddress = generateStreetAddress()
+            expect(typeof streetAddress).toEqual('string')
+            expect(streetAddress).toMatch(/^\d+ [A-Za-z]+(?: [A-Za-z]+)?$/)
```

Comment:
Can we either call this many times in a loop, or put it through fast-check? the fast-check test should look almost identical since there are no inputs - but it would be good to see these running more than once in tests.

Reply:
Yep, fair point. Will do.

Reply:
Actually, in this instance, for 'generateStreetAddress' fast-check won't be any better than a loop

Reply:
Added in [https://github.com/duckduckgo/content-scope-scripts/pull/1021/commits/c2a9e1de9f8fbf42d4c8dfa7436e3b9d4e6515dc](c2a9e1de9f8fbf42d4c8dfa7436e3b9d4e6515dc)